### PR TITLE
[RFC] node/provider: allow specified node host prog

### PR DESCRIPTION
**Pull request by chemzqm**
Problem: The system call `npm root -g` could take more than 300ms on my local machine, it's quite slow!

Result of `nvim --startuptime`
```
433.311  331.646  331.646: sourcing /usr/local/share/nvim/runtime/autoload/provider/node.vim
```

Result from fish shell
```
~/lib/neovim(improve_node_provider|✔) 
❯ npm root -g
/usr/local/lib/node_modules

~/lib/neovim(improve_node_provider|✔)
❯ echo $CMD_DURATION
350
```

Solution: Provide `g:node_host_prog` like ruby provider so that the user could speed up their neovim.

### CODE REVIEW

**Comment by justinmk** [comment]
Thanks, `provider.txt` help needs an update too.

File: runtime/doc/provider.txt [hunk 1]
```diff
@@ -112,6 +112,32 @@ To use an absolute path (e.g. to an rbenv installation): >
...
+To use javascript remote plugins with Nvim, just install the latest `neovim`
+ package: >
+    $ npm install -g neovim
+
+NODE PROVIDER CONFIGURATION~
+
+						*g:loaded_node_provider*
+To disable Node support: >
+    let g:loaded_node_provider = 1
+<
```
Comment (mhinz) [inline runtime/doc/provider.txt]: Here you close the `helpExample`, but not in the other cases.

Comment (chemzqm) [reply runtime/doc/provider.txt]: fixed

**Comment by justinmk** [comment]
merged

**Comment by janlazo** [comment]
`let g:node_host_prog = '/usr/local/bin/neovim-node-host'` changes the semantics of that function because that's the shim. It should be the absolute path of node host's `bin/cli.js`.

**Pull request closed without merging**